### PR TITLE
Feature/contract negotiation integration tests

### PR DESCRIPTION
**Pull request by davidjovanovic**

### CODE REVIEW

**Review by marest94 — APPROVED** [review]

File: connector/src/test/java/it/eng/connector/integration/CatalogIntegrationTest.java [hunk 1]
```diff
@@ -15,94 +17,85 @@
...
-    private DatasetRequestMessage datasetRequestMessage = DatasetRequestMessage.Builder.newInstance()
-            .dataset(Serializer.serializeProtocol(MockObjectUtil.DATASET)).build();
-
-    // this can be found in the initial_data.json
-    private final String DATASET_ID = "fdc45798-a222-4955-8baf-ab7fd66ac4d5";
-
+    
     @Test
-    @WithUserDetails("milisav@mail.com")
+    @WithUserDetails(TestUtil.USER_DETAILS)
```
Comment (IgorBalog-Eng) [inline connector/src/test/java/it/eng/connector/integration/CatalogIntegrationTest.java:33]: Nice one :)

File: doc/test_containers_starting_guide.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+This is a starting reference for adding test containers to the project (probably needed when we add DAPS to integration tests).
```
Comment (IgorBalog-Eng) [inline doc/test_containers_starting_guide.md:1]: Could add link to create custom Testcontainer
https://java.testcontainers.org/features/creating_container/

File: catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java [hunk 3]
```diff
@@ -14,28 +15,29 @@
...
 public class CatalogExceptionAdvice extends ResponseEntityExceptionHandler {
 
     @ExceptionHandler(value = {CatalogErrorException.class})
-    protected ResponseEntity<String> handleCatalogErrorException(CatalogErrorException ex) {
+    protected ResponseEntity<Object> handleCatalogErrorException(CatalogErrorException ex) {
         CatalogError catalogError = CatalogError.Builder.newInstance()
         		.code(HttpStatus.NOT_FOUND.getReasonPhrase())
                 .reason(Arrays.asList(Reason.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build()))
                 .build();
+//        return handleExceptionInternal(ex, response, new HttpHeaders(), HttpStatus.NOT_FOUND, request);
```
Comment (IgorBalog-Eng) [inline catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java:29]: handleExceptionInternal or  return new ResponseEntity<>?

File: doc/test_containers_starting_guide.md (hunk 2 above)
Comment (IgorBalog-Eng) [reply doc/test_containers_starting_guide.md:1]: Disregard this comment, I did not go all the way to the end of document

File: connector/src/test/java/it/eng/connector/integration/CatalogIntegrationTest.java (hunk 1 above)
Comment (davidjovanovic) [reply connector/src/test/java/it/eng/connector/integration/CatalogIntegrationTest.java:33]: Thanks 👍

File: doc/test_containers_starting_guide.md (hunk 2 above)
Comment (davidjovanovic) [reply doc/test_containers_starting_guide.md:1]: 👍

File: catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java (hunk 3 above)
Comment (davidjovanovic) [reply catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java:29]: yep, we agreed to use handleExceptionInternal, I forgot to change it here too, good that you found it :)

File: negotiation/src/main/java/it/eng/negotiation/exception/ContractNegotiationExceptionAdvice.java [hunk 4]
```diff
@@ -25,21 +30,37 @@ protected ResponseEntity<Object> handleContractNotFound(ContractNegotiationNotFo
...
         ContractNegotiationErrorMessage errorMessage = ContractNegotiationErrorMessage.Builder.newInstance()
                 .providerPid(ex.getProviderPid())
-                .code(HttpStatus.NOT_FOUND.getReasonPhrase())
+                .code(HttpStatus.CONFLICT.getReasonPhrase())
                 .reason(Collections.singletonList(Reason.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build()))
                 .description(Collections.singletonList(Description.Builder.newInstance().language("en").value(ex.getLocalizedMessage()).build())).build();
-        String response = Serializer.serializeProtocol(errorMessage);
 
-        return handleExceptionInternal(ex, response, new HttpHeaders(), HttpStatus.NOT_FOUND, request);
+        return handleExceptionInternal(ex, Serializer.serializeProtocolJsonNode(errorMessage), new HttpHeaders(), HttpStatus.CONFLICT, request);
```
Comment (IgorBalog-Eng) [inline negotiation/src/main/java/it/eng/negotiation/exception/ContractNegotiationExceptionAdvice.java:60]: Is this conflict status according to DSP specification?

File: catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java [hunk 5]
```diff
@@ -2,44 +2,40 @@
...
 import it.eng.catalog.model.CatalogError;
-import it.eng.catalog.model.GenericApiResponse;
 import it.eng.catalog.model.Reason;
 import it.eng.catalog.model.Serializer;
 import it.eng.catalog.rest.protocol.CatalogProtocolController;
-import it.eng.catalog.service.CatalogService;
 import jakarta.validation.ValidationException;
 
-@RestControllerAdvice(basePackageClasses = {CatalogProtocolController.class, CatalogService.class})
+@RestControllerAdvice(basePackageClasses = CatalogProtocolController.class)
```
Comment (IgorBalog-Eng) [inline catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java:19]: With excluding Protocol from names in classes, should we do the same with Controller?
And use API in all other cases where non-protocol code is involved.

File: connector/src/test/java/it/eng/connector/util/TestUtil.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package it.eng.connector.util;
+
+public class TestUtil {
+	//all values in this class are from the initial_data.json
+	public static final String USER_DETAILS = "connector@mail.com";
```
Comment (IgorBalog-Eng) [inline connector/src/test/java/it/eng/connector/util/TestUtil.java:5]: Can you also rename this one to CONNECTOR_USER?

Comment (davidjovanovic) [reply connector/src/test/java/it/eng/connector/util/TestUtil.java]: No problemo 👍

File: catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java (hunk 5 above)
Comment (davidjovanovic) [reply catalog/src/main/java/it/eng/catalog/exceptions/CatalogExceptionAdvice.java]: Done 👍

File: negotiation/src/main/java/it/eng/negotiation/exception/ContractNegotiationExceptionAdvice.java (hunk 4 above)
Comment (davidjovanovic) [reply negotiation/src/main/java/it/eng/negotiation/exception/ContractNegotiationExceptionAdvice.java]: You were right, I changed it

**Pull request merged**